### ISSUE TITLE
Mesecons upgrade request 

### ISSUE THREAD

**Issue by Klaranth** [pandorabox-io/in-game#290]
((Hedgehog : Added WIP label, as this might already be in the works. @OgelGames - please check for this) 
@dennisjenkins75 ; @OgelGames ; @S-S-X ; @BuckarooBanzay

1hittoaster 
Please review / approve https://github.com/mt-mods/mesecons/pull/7
Also, would be really nice to update the mt-mods/mesecons mod on pandorabox once the above is merged into upstream.

@OgelGames 
There were plans to do that, but we ended up deciding that the new features need some work before it can be used on pandorabox
see https://github.com/pandorabox-io/pandorabox-mods/pull/2082 and https://github.com/mt-mods/mesecons/pull/6

1hittoaster 
Is there an objection to adding the "pos" global? I modelled it after how "heat" was implemented. Tested locally, works.

OgelGames
No, thats a great addition

1hittoaster 
its a lot more effecient than asking a JD for its coords, and maintaining state, waiting for the response, etc...

OgelGames 
although i think it might need to be copied first

1hittoaster — 06/09/2022
My only concern was should it be called "pos", or something more unique, like "luac_pos" ?
My primitive testing showed that it does work, but you're right, maybe it needs to be copied.
I did not try to modify it to see if that could be abused to edit meta of some other luac.
like find the coords of someone else's luac, then in yours do "pos = other_players_pos" and then try to do nasty things.
feedback on the PR is welcome.  I can edit it tomorrow.  I'm about done for the night.

SX
About mt-mods mesecons: not sure if it is good idea but I think it might be good idea to start over:
break out mt-mods master to dev branch, reset mt-mods master branch to latest minetest-mods master, cherry-pick bug fixes and simple features from mt-mods dev branch and call it a day.

Then start picking up other additional features and deciding where those belong, some stuff directly to mt-mods mesecons and some probably to separate mod. Like luacontroller examples/terminal/themes/possibly some other feature updates and probably pulse gate.
mt-mods version is not really in use anywhere yet right? or is it?
If it is could still do that in another branch without breaking current master, that would probably be best option to be able to actually use at least some new features and bug fixes.

wsor 
Instead of forking. Just monkey patch/override what you want in.

SX
That did not seem to be possible with mesecons updates as it would mean overriding mods almost completely because changes have to be done for internal stuff and API added for extensions. I don't think even that simple pos function could be added easily without forking whole thing and other changes are a lot larger.

BuckarooBanzay
It is afaik in use on some servers from VanessaE

Most changes were to the luacontroller, how about creating a new luacontroller mod that coexists with the existing one and has all the enhancements? (console, library support, etc)
Upsides:
* Our own lifecycle
* Standalone mod (does one thing and that properly)
* No conflict to existing mesecons-mod (only the actionqueue as an interface)
Downsides:
* ?

1hittoaster
Honestly, other than the "lua sorting tube", I am surprised about the mixing of digiline, mesecon and luac stuff in that one mod.  Seems like a "kitchen sink" mod.  Splitting the LUAC and lua-tube into its own mod seems reasonable.

SX 
Luacontroller is already in its own mod but just included in mesecons modpack, for lua tube yes it seems like separate mod could be better. For mesecons modpack is really nice way to distribute it, I think even while there's not that much content mods similar would still be good for pipeworks. Also for technic (larger content modpack with more machines and tools).
For pipeworks as it is not modpack itself it would be best to build modpack with git submodules.

**Comment by OgelGames** [comment]
The current state is basically an agreement that it needs to be improved, and will be, but nothing has actually been done yet.

I really need to put some time aside to write down my lua-sandbox-api-mod concept so it can be can discussed better. Now that I'm about done with redoing bones, I might do that soon-ish.

**Comment by Klaranth** [comment]
@OgelGames  Does that mean, I should assign you to this issue? :)

**Comment by OgelGames** [comment]
If you want, doesn't really make any difference for me.

**Comment by Klaranth** [comment]
Huhhila
@dennisjenkins75 : also #282 is related; use vector.copy(pos) ?

**Comment by Klaranth** [comment]
The4spaceconstants
Reshaping 18 mesecon into a mese crystal

**Comment by SwissalpS** [comment]
> The4spaceconstants Reshaping 18 mesecon into a mese crystal

Assuming you want to convert mesecon-wires back into mese-crystal: A recipe wouldn't be enough, there was heat involved in making wires, converting back should at minimum also involve heat. (preferably also some process like slow cooling or tempering to regain crystal form)
